From 49813bca064ec8311745560e5ecc1797f7cc9f61 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Thu, 11 Sep 2014 12:00:58 -0400
Subject: [ethernet] mlx4: Deprecate use_prio module parameter

Message-id: <1fb49a9c0408a843124ace5b67d6f220cf022176.1410425016.git.avadai@redhat.com>
Patchwork-id: 93659
O-Subject: [Patch RHEL7.1 V1 122/197] net/mlx4_core: Deprecate use_prio module parameter
Bugzilla: 1107617 1107618 1107619 1107620
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Amir Vadai <amirv@mellanox.com>

use_prio was added as part of an infrastructure for running FCoE in A0 mode.
FCoE didn't get into Mellanox Upstream driver, and when it will, it won't be
using A0 steering mode.

Therefore we can safely deprecate this module parameter without hurting any
existing user.

CC: Carol Soto <clsoto@linux.vnet.ibm.com>
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit ecc8fb11cdb37d108d4597ba0f6bdff77c6019af)
Signed-off-by: Amir Vadai <avadai@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/main.c b/drivers/net/ethernet/mellanox/mlx4/main.c
index 526cca8..8b1c2c1 100644
--- a/drivers/net/ethernet/mellanox/mlx4/main.c
+++ b/drivers/net/ethernet/mellanox/mlx4/main.c
@@ -132,8 +132,7 @@ MODULE_PARM_DESC(log_num_vlan, "Log2 max number of VLANs per ETH port (0-7)");
 
 static bool use_prio;
 module_param_named(use_prio, use_prio, bool, 0444);
-MODULE_PARM_DESC(use_prio, "Enable steering by VLAN priority on ETH ports "
-    "(0/1, default 0)");
+MODULE_PARM_DESC(use_prio, "Enable steering by VLAN priority on ETH ports (deprecated)");
 
 int log_mtts_per_seg = ilog2(MLX4_MTT_ENTRY_PER_SEG);
 module_param_named(log_mtts_per_seg, log_mtts_per_seg, int, 0444);
@@ -290,7 +289,6 @@ static int mlx4_dev_cap(struct mlx4_dev *dev, struct mlx4_dev_cap *dev_cap)
 
  dev->caps.log_num_macs  = log_num_mac;
  dev->caps.log_num_vlans = MLX4_LOG_NUM_VLANS;
- dev->caps.log_num_prios = use_prio ? 3 : 0;
 
  for (i = 1; i <= dev->caps.num_ports; ++i) {
   dev->caps.port_type[i] = MLX4_PORT_TYPE_NONE;
@@ -358,7 +356,6 @@ static int mlx4_dev_cap(struct mlx4_dev *dev, struct mlx4_dev_cap *dev_cap)
   dev->caps.reserved_qps_cnt[MLX4_QP_REGION_FC_ADDR] =
   (1 << dev->caps.log_num_macs) *
   (1 << dev->caps.log_num_vlans) *
-  (1 << dev->caps.log_num_prios) *
   dev->caps.num_ports;
  dev->caps.reserved_qps_cnt[MLX4_QP_REGION_FC_EXCH] = MLX4_NUM_FEXCH;
 
@@ -2775,6 +2772,9 @@ static int __init mlx4_verify_params(void)
   pr_warning("mlx4_core: log_num_vlan - obsolete module param, using %d\n",
       MLX4_LOG_NUM_VLANS);
 
+ if (use_prio != 0)
+  pr_warn("mlx4_core: use_prio - obsolete module param, ignored\n");
+
  if ((log_mtts_per_seg < 1) || (log_mtts_per_seg > 7)) {
   pr_warning("mlx4_core: bad log_mtts_per_seg: %d\n", log_mtts_per_seg);
   return -1;
diff --git a/include/linux/mlx4/device.h b/include/linux/mlx4/device.h
index 333ae3a..fd365e4 100644
--- a/include/linux/mlx4/device.h
+++ b/include/linux/mlx4/device.h
@@ -449,7 +449,6 @@ struct mlx4_caps {
  int                     reserved_qps_base[MLX4_NUM_QP_REGION];
  int                     log_num_macs;
  int                     log_num_vlans;
- int                     log_num_prios;
  enum mlx4_port_type port_type[MLX4_MAX_PORTS + 1];
  u8   supported_type[MLX4_MAX_PORTS + 1];
  u8                      suggested_type[MLX4_MAX_PORTS + 1];
-- 
1.7.1